### PR TITLE
Move selected rows state of Data Catalog to zustand store

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Eliminates lag when using drag-to-select (the `SelectionBrush`) by moving the state into a zustand store.
  - This avoids the lag because now the entire Data Catalog no longer has to rerender, because the state is no longer stored in the `AssetsTable` component that contains all the rows (and would therefore rerender all the rows when its state changes)

### Important Notes
- The lag is present on Chromium, but any lag in general is generally more visible on Firefox, so it's highly recommended to test on Firefox as well as Electron
  - On current develop, *any* drag selection should be enough to trigger the lag (typically 200ms JS + 200ms rendering). If it's not reproducible, then you may need to create more assets.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

### CODE REVIEW

**Comment by PabloBuchu** [comment]
QA ✅ LGTM

**Review by PabloBuchu — APPROVED** [review]

File: app/dashboard/src/layouts/AssetsTable.tsx [hunk 1]
```diff
@@ -2256,37 +2254,27 @@ export default function AssetsTable(props: AssetsTableProps) {
...
-                event.stopPropagation()
-                setMostRecentlySelectedIndex(visibleItems.indexOf(item))
-                selectionStartIndexRef.current = null
-                setSelectedKeys(new Set([key]))
-              }
+            select={() => {
+              setMostRecentlySelectedIndex(visibleItems.indexOf(item))
+              selectionStartIndexRef.current = null
+              setSelectedKeys(new Set([item.key]))
             }}
```
Comment (indiv0) [inline app/dashboard/src/layouts/AssetsTable.tsx:2269]: Where did this `onContextMenu` behaviour go?

**Review by indiv0 — APPROVED** [review]
LGTM but admittedly I'm a bit out of my element when it comes to React state management

File: app/dashboard/src/layouts/AssetsTable.tsx (hunk 1 above)
Comment (somebody1234) [reply app/dashboard/src/layouts/AssetsTable.tsx:2269]: inlined into `AssetRow.tsx` because the selected state is now being handled in each row

**Pull request merged**